src/gen.ts

```diff
@@ -74,10 +74,12 @@ export class Generator {
     private checkers: Checker[] = [];
     private header: string | null;
     private boundedRecRules: Set<string>;
+    private astKindsByName: Map<string, string> = new Map();
```

Comment:
I think it's not ideal that this is a class-level field, but shouldn't be accessed until `writeKinds` is called. Could we move the computation of `astKinds` from the `writeKinds` method to the constructor so that this map can also be filled out by the constructor?